### ISSUE TITLE
Phoenix has exposed dependency to linc_opengl

### ISSUE THREAD

**Issue by NtrfAtZNS** [underscorediscovery/luxe#368]
I'm probably misusing luxe as much as possible, but i'm using it for 3d multiplayer game with dedicated server. As you might imagine, luxe (as well as snow and flow) are not very cooperative when being compiled without OpenAL, OpenGL, SDL while targeting ... Java VM. Yes, i'm compiling it for Java VM with custom main class and hand-written .hxml file. I had to copy parts of generated .hxml into a custom .hxml scince flow does not support such target (although i still use flow for native dedicated server). I had to fork snow in order to remove dependency on SDL, OpenAL and other native libs and fix some bugs that made it incompatible with java target. However in luxe i only had to change one line to make it work:

```diff
diff --git a/phoenix/Renderer.hx b/phoenix/Renderer.hx
index 001624b..a1d4211 100644
--- a/phoenix/Renderer.hx
+++ b/phoenix/Renderer.hx
@@ -311,5 +311,5 @@ class Renderer {
         var frag_bitmapfont = null;

-        #if (!linc_opengl_GLES && !js)
+        #if (!linc_opengl_GLES && !js && linc_opengl)
             if(Luxe.snow.config.render.opengl.profile == snow.types.Types.OpenGLProfile.core) {
                 var vaos = [0];
```

In short - even if platform is not `js` it does not mean `linc_opengl` library is available. I hope i'm not asking for much.

**Comment by underscorediscovery** [comment]
You should have had to fork snow to change the modules,  you can specify even the empty internal modules and it will allow it. You can also implement a headless runtime of your own in a few lines of code (like others have been for dedicated server). I'd be curious why you a fork was necessary to just specify new modules.

That new code in phoenix is temporarily placed and wasn't shielded properly, I pushed a fix now for it thanks.

**Comment by NtrfAtZNS** [comment]
I forked snow because it was the fastest solution. My primary focus for now is network protocol for my game and i didn't want to waste time on other tasks now.

I'm not sure how "others" have been doing it with "a few lines of code", but my server is a linux machine without OpenGL and X11 libraries. This means that server version has to be compiled without SDL, OpenGL and OpenAL dependencies. I've tried to invoke macro `snow.Set.*` to change runtime class, but SDL runtime is still compiled, because there is a `Compiler.keep` call. Even after i've made a fork and reconfigured classes, c++ part of SDL is still being compiled.

This is where i decided to switch to Java target. It's easier to deploy and it did not require a lot of modification. In the end i have a horrible mess, that i will remove entirely as soon as my own game engine is ready (i'm waiting for haxe 3.3 rc).

**Comment by underscorediscovery** [comment]
I am aware of the requirements. Here's what I meant with the few lines:

![screen 2016-03-28 at 2 44 50 pm](https://cloud.githubusercontent.com/assets/236374/14084088/c26ea342-f4f3-11e5-832c-03c2b0a73552.png)

The modules in snow are designed to be configured for situations like this. The specifics of the build part are the real problem are described below.

> but SDL runtime is still compiled, because there is a `Compiler.keep`

The reason is actually because of `framework.flow` dependencies, flow `dependencies` node currently can't be declared under an `if` node because they were made to be deterministic at the time.

For now, I pushed some fixes to the empty modules that were missing since the alpha update, it should allow you to configure away the _code_ dependency, then the only thing would be to edit `snow/framework.flow` for the _build_ dependency in your fork to get away from linc libs being added.

I'm already well underway solving the build dependencies with flow alpha updates, as with all the alpha stuff, it will get there but we take one step at a time. 

Thanks for elaborating.
